url29ecd5d728_geodb_db3dcore_tgz-pJ8-de_uos_igf_db3d_dbms_model3d_SegmentNetBuilderJ8/src/de/uos/igf/db3d/resources/DB3DProperties.java

```
@@ -0,0 +1,145 @@
+	private static File propertiesFile;
+
+	static {
+
+		/*
+		 * CAUTION!: do not use logging in this static constructor, since the
+		 * logging architecture of Db3dLogger itself depends on an object of
+		 * this class. Simply use printStackTrace() method instead.
+		 */
+
+		try {
+
+//			String appFolderString = DB3DProperties.class.getResource(
+//					DB3DProperties.class.getSimpleName() + ".class").toString();
+//			
+//			System.out.println(":::::::::::::::::::::::::::::::::: "+appFolderString);
+//
+//			/*
+//			 * cut
+//			 * "file:/.../DB3DoDB/bin/de/uos/igf/db3d/resources/Db3dProperties.class"
+//			 * to ".../DB3DoDB/":
+//			 */
+//			appFolderString = appFolderString.substring(5, appFolderString
+//					.length()
+//					- (DB3DProperties.class.getCanonicalName().length() + 10));
+//
+//			String propertiesFolderString = appFolderString + "properties";
+			
+			String propertiesFolderString = "etc/";
+			
+			File propertiesFolder = new File(propertiesFolderString);
+
+			if (!propertiesFolder.exists()) {
+				propertiesFolder.mkdir();
+			}
+
+			propertiesFile = new File(propertiesFolderString
+					+ "/db3d.properties");
+
+			if (!propertiesFile.exists()) {
+				propertiesFile.createNewFile();
+			}
+
+			if (propertiesFile.exists()) {
+				db3dProperties.load(new FileInputStream(propertiesFile));
+			}
+
+		} catch (Exception ex) {
+			// Db3dLogger can not be used here, since the logger needs the
+			// Db3dProperties, so:
+			System.out.println("DB3DProperties: Can not find db3d.properties file "
+					+ "in properties/ folder of the server.");
+		}
+
+	}
+
+	// prohibit instantiation:
+	private DB3DProperties() {
+	}
+
+	public static void store(String comments) throws IOException {
+		if (propertiesFile != null && propertiesFile.exists()) {
+			try {
+				FileOutputStream fw = new FileOutputStream(propertiesFile);
+				db3dProperties.store(fw, comments);
+			} catch (IOException e) {
```

Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)